Cython/Compiler/PyrexTypes.py

```diff
@@ -4458,20 +4458,17 @@ def error_condition(self, result_code):
         return "dummy"
 
 
-class PythonTypeConstructor(PyObjectType):
+class PythonTypeConstructorMixin(object):
     """Used to help Cython interpret indexed types from the typing module (or similar)
     """
     modifier_name = None
 
-    def __init__(self, name, base_type=None):
+    def __init__(self, name):
         self.python_type_constructor_name = name
```

Comment:
Given your usage below, it seems better to avoid `__init__` here and use an explicit initialisation method, e.g. `set_python_type_constructor_name()`. Then you can call that below and use `super()` for the normal constructor chain.